### PR TITLE
Do not wait to retry when configLeader exists

### PR DESCRIPTION
**Pull request by Pengzna**
After introducing https://github.com/apache/iotdb/pull/11941, when the configNode client is newly created and looking for a leader, it is expected to not wait if there is a leader.

Detailed docs：https://apache-iotdb.feishu.cn/docx/XlHkdnkQvohOCwxAduJc0rutnNh

### CODE REVIEW

**Review by CRZbulabula — APPROVED** [review]
LGTM!

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/protocol/client/ConfigNodeClient.java [hunk 1]
```diff
@@ -243,6 +226,15 @@ private void tryToConnect() throws TException {
         logger.warn("The current node may have been down {},try next node", configLeader);
         configLeader = null;
       }
+    } else {
+      try {
+        // Wait to start the next try
+        Thread.sleep(RETRY_INTERVAL_MS);
+      } catch (InterruptedException e) {
+        Thread.currentThread().interrupt();
+        throw new TException(
```
Comment (liyuheng55555) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/protocol/client/ConfigNodeClient.java:235]: Seems not necessary to throw an exception here ?

**Review by liyuheng55555 — CHANGES_REQUESTED** [review]

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/protocol/client/ConfigNodeClient.java (hunk 1 above)
Comment (Pengzna) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/protocol/client/ConfigNodeClient.java]: fixed

**Review by liyuheng55555 — APPROVED** [review]
LGTM~

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**